src/components/GoodsList/GoodsList.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Good } from '../../types/Good';
+import './GoodsList.scss';
+
+interface Props {
+  goods: Good[],
+}
+
+interface State {
+  isReversed: boolean,
+  sortBy: string,
+  length: number,
+}
+
+export class GoodsList extends React.PureComponent<Props, State> {
+  state: State = {
+    isReversed: false,
+    sortBy: '',
+    length: 1,
+  };
+
+  reverse = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState({ sortBy: 'alphabet' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({ sortBy: '', length: 1 });
+  };
+
+  changeLength = (value: number) => {
+    this.setState({ length: value });
+  };
+
+  render() {
+    const { sortBy, isReversed, length } = this.state;
+    const goods = [...this.props.goods].filter(el => el.length > length);
```

Comment:
```suggestion
    const goods = this.props.goods.filter(el => el.length > length);
```
Метод filter сам створює копію масиву, тому spread-оператор тут надлишковий.